src/Nethermind/Nethermind.Optimism/Rpc/OptimismEthRpcModule.cs

```diff
@@ -235,7 +251,7 @@ protected override ResultWrapper<TransactionForRpc> GetTransactionByBlockAndInde
                     txIndex: index);
                 if (transactionModel is DepositTransactionForRpc depositTx)
                 {
-                    OptimismTxReceipt? receipt = receipts.FirstOrDefault(r => r.TxHash?.Equals(hash) ?? false);
+                    OptimismTxReceipt? receipt = (OptimismTxReceipt?)receipts.FirstOrDefault(r => r.TxHash?.Equals(hash) ?? false);
```

Comment:
Could you provide a comment or spec link why is this always the case?

Reply:
According to the yellow paper, the Receipt type is the same as the Transaction type, so a Deposit transaction should be associated with a Deposit receipt.

If you take a look at op-geth's codebase, when working with a Deposit transaction, the associated receipt contains Optimism's specific fields (ex. `DepositNonce`), which matches our `OptimismTxReceipt` type.

https://github.com/ethereum-optimism/op-geth/blob/d1b66c4722e7fb5525efbcadda4c1ab3f7dacaa8/internal/ethapi/api.go#L1611-L1616

Unfortunately, a lot of these invariants are not part of the type system and I haven't been able to find them expressed in the specs so I might be wrong about them.